4-WebApp-your-API/4-1-Your-API/README.md

```diff
@@ -101,368 +79,348 @@ There are two projects in this sample. Each needs to be separately registered in
+  - set the authority to be the Microsoft identity platform v2.0 identity
+  - sets the audiences to validate
+  - register an issuer validator that accepts issuers to be in the Microsoft identity platform clouds.
 
-The code for this method is available in [Microsoft.Identity.Web\Client\TokenAcquisition.cs L394-L427](https://github.com/Azure-Samples/active-directory-dotnet-native-aspnetcore-v2/blob/2e5d34aedabbf20b583580a661433ad4b00bb035/Microsoft.Identity.Web/Client/TokenAcquisition.cs#L394-L427)
+The implementations of these classes are in the Microsoft.Identity.Web library (and folder), and they are designed to be reusable in your applications (Web apps and Web apis). You are encouraged to browse the code in the library to understand the changes in detail.
 
-#### On the client side
+- At the beginning of the `Configure` method, insert `app.UseSession()`. This code ensures that the session exists for the session-based token cache to work properly. You can skip this if you do not plan to use the session based token cache.
 
-On the client side, when it calls the Web API and receives a 403 with a www-Authenticate header, the client will call the `HandleChallengeFromWebApi` method, which will
+  ```CSharp
+  public void Configure(IApplicationBuilder app, IHostingEnvironment env)
+  {
+    app.UseSession();
+    ...
+  ```
 
-- extract from the www-Authenticate header
-  - the client ID
-  - the scopes
-  - the claims
-- Create a new `PublicClientApplication` (as the client is here a desktop app), and call `AcquireTokenAsync` method to ask the user to consent.
+### Create the TodoListController.cs file.
 
-The code for `HandleChallengeFromWebApi` method is available from [TodoListClient\MainWindow.xaml.cs L180-L206](https://github.com/Azure-Samples/active-directory-dotnet-native-aspnetcore-v2/blob/9be6d8b3705ada352caa9a47be0a5add8389b6f6/2.%20Web%20API%20now%20calls%20Microsoft%20Graph/TodoListClient/MainWindow.xaml.cs#L180-L206)
+1. Add a folder named `Models` and then create a new  file named `TodoItem.cs`. Copy the contents of the TodoListClient\Models\TodoItem.cs in this file.
+1. Create a new Controller named `TodoListController` and copy and paste the code from the sample (\TodoListService\Controllers\TodoListController.cs) to this controller.
 
 ## How to deploy this sample to Azure
 
-See section [How to deploy this sample to Azure](../1.%20Desktop%20app%20calls%20Web%20API/README.md#How-to-deploy-this-sample-to-Azure) in the first part of this tutorial, as the deployment is exactly the same.
+This project has two WebApp / Web API projects. To deploy them to Azure Web Sites, you'll need, for each one, to:
+
+- create an Azure Web Site
+- publish the Web App / Web APIs to the web site, and
+- update its client(s) to call the web site instead of IIS Express.
+
+### Create and publish the `TodoListService-aspnetcore-webapi` to an Azure Web Site
```

Comment:
I have noticed that previous steps in this tutorial doesn't cover how to deploy to Azure, since this is done on step 6. I would suggest removing this section from this Readme, to keep it consistent with the other samples here. 
What do you think?

Reply:
I have removed it from the README-incremental-instructions.md, but left in readme.md

---
In reply to: [297416850](https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/pull/119#discussion_r297416850) [](ancestors = 297416850)